src/ddmd/parse.d

```diff
@@ -7770,6 +7770,10 @@ final class Parser(AST) : Lexer
         case TOKdelete:
             nextToken();
             e = parseUnaryExp();
+            // 1. Deprecation for 1 year
+            // 2. Error for 1 year
+            // 3. Removal, "delete" can be used for other identities
```

Comment:
How about using a similar deprecated pattern as in Phobos?
```
// @@@DEPRECATED_2017-12@@@
```

In any case, it would be good to start using easily greppable years into the source code, s.t. these deprecations don't get forgotten.

Reply:
> // @@@DEPRECATED_2017-12@@@

 How can I be sure that this will be merged this month?

Reply:
The trick is to annoy @andralex and @WalterBright daily ;-)
Okay all jokes aside, at Phobos these deprecation strings are put in a PR and this has never been a huge problem as at least for Phobos deprecation PRs are merged rather quickly. Yep, there have been cases in which a PR has been merged later, but it hasn't been a big deal whether a deprecation lasted for 23 or 24 months. On the contrary (1) with Martin's polished release two months release cycle, it's still 9 or 10 releases (compared to 3-4 releases two or three years ago) and people are now complaining that the deprecation periods are too long. And secondly (2) the obvious advantage is that someone will actually switch the deprecation to an error.
Anyhow, the deprecation cycle is rather ill-defined as of now.

Reply:
OK, I added `@@@DEPRECATED_2017-12@@@`.